src/elections/urls.py

```diff
@@ -0,0 +1,160 @@
+    tags=["elections"],
+)
+
+def _slugify(
+        text: str
+) -> str:
+    """
+    Creates a unique slug based on text passed in. Assumes non-unicode text.
+    """
+    return re.sub(r"[\W_]+", "-", text)
+
+async def _validate_user(
+        db_session: database.DBSession,
+        session_id: str
+) -> dict:
+    computing_id = await auth.crud.get_computing_id(db_session, session_id)
+    # Assuming now user is validated
+    result = await types.ElectionOfficer.has_permission(db_session, computing_id)
+    return result
+
+@router.get(
+    "/create_election",
+    description="Creates an election and places it in the database",
+)
+async def create_election(
+    request: Request,
+    db_session: database.DBSession,
+    name: str,
+    election_type: str,
+    start_datetime: datetime | None = None,
+    end_datetime: datetime | None = None,
+    survey_link: str | None = None
+):
+    """
+    aaa
+    """
+    session_id = request.cookies.get("session_id", None)
+    user_auth = await _validate_user(db_session, session_id)
+    if user_auth is False:
+        # let's workshop how we actually wanna handle this
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="You do not have permission to access this resource",
+            headers={"WWW-Authenticate": "Basic"},
+        )
+
+    # Default start time should be now unless specified otherwise
+    if start_datetime is None:
+        start_datetime = datetime.now()
+
+    if election_type not in election_types:
+        raise RequestValidationError()
+
+    params = ElectionParameters(
+        _slugify(name),
+        name,
+        await auth.crud.get_computing_id(db_session, session_id),
+        election_type,
+        start_datetime,
+        end_datetime,
+        survey_link
+    )
+
+    await elections.crud.create_election(params, db_session)
+    await db_session.commit()
+
+    # TODO: create a suitable json response
+    return {}
+
+@router.get(
+        "/delete_election",
+        description="Deletes an election from the database"
+)
+async def delete_election(
+    request: Request,
+    db_session: database.DBSession,
+    slug: str
+):
+    session_id = request.cookies.get("session_id", None)
+    user_auth = await _validate_user(db_session, session_id)
+    if user_auth is False:
+        # let's workshop how we actually wanna handle this
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="You do not have permission to access this resource",
+            headers={"WWW-Authenticate": "Basic"},
+        )
+
+    if slug is not None:
+        await elections.crud.delete_election(slug, db_session)
+        await db_session.commit()
+
+@router.get(
+    "/update_election",
+    description="""Updates an election in the database.
+                   Note that this does not allow you to change the _name_ of an election as this would generate a new slug."""
+)
```

Comment:
```suggestion
@router.patch(
    "/election/{slug:str}",
    description="""Updates an election in the database.
                   Note that this does not allow you to change the _name_ of an election as this would generate a new slug."""
)
```